src/v/coproc/event_listener.cc

```diff
@@ -70,7 +70,7 @@ ss::future<> event_listener::start() {
           return type_and_handler.second->start();
       });
 
-    (void)ss::with_gate(_gate, [this] {
+    ssx::spawn_with_gate(_gate, [this] {
```

Comment:
I wonder if it's worth moving the main exception handler out, to ignore gate closed as well as source aborted? @graphcareful

Reply:
The way I read this code, it's intending to keep running `do_start` again and again, even if there's an exception -- if we moved the exception handler outside spawn_with_gate, then any exception in `do_start` would cause us to drop out of the `do_until` and not retry.